src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Markup/Baml2006/Baml2006Reader.cs

```diff
@@ -2065,15 +2065,15 @@ private string Logic_GetFullyQualifiedNameForType(XamlType type)
 
         private string Logic_GetFullXmlns(string uriInput)
         {
-            int colonIdx = uriInput.IndexOf(":");
+            int colonIdx = uriInput.IndexOf(':');
             if (colonIdx != -1)
             {
                 string uriTypePrefix = uriInput.Substring(0, colonIdx);
                 if (String.Equals(uriTypePrefix, "clr-namespace"))
                 {
                     //We have a clr-namespace so do special processing
                     int clrNsStartIdx = colonIdx + 1;
-                    int semicolonIdx = uriInput.IndexOf(";");
+                    int semicolonIdx = uriInput.IndexOf(';');
                     if (-1 == semicolonIdx)
                     {
                         // We need to append local assembly
```

Comment:
The condition should be refactored to avoid one intermediate string allocation:
```
return (_settings.LocalAssembly != null)
  ? uriInput + ";assembly=" + GetAssemblyNameForNamespace(_settings.LocalAssembly)
  : uriInput;